ui/src/models/artifactTypes.model.ts

```diff
@@ -21,7 +21,11 @@ import { Services } from "src/services";
 export class ArtifactTypes {
 
     public static async allTypes(): Promise<string[]> {
-        return (await Services.getAdminService().getArtifactTypes()).map(t => t.name);
+        try {
+            return (await Services.getAdminService().getArtifactTypes()).map(t => t.name);
```

Comment:
Does this actually throw an exception or can this return a failed promise?

Reply:
this should take into account both scenarios:
- a failed `Promise` failed to `map` and get expanded to an Exception
- a direct Exception will be caught as usual